website/docs/tutorials/07-testing/02-cheatcodes/example.md

```diff
@@ -0,0 +1,43 @@
+# `example`
+
+```python
+def example(self, *args: Any, **kwargs: Any) -> None:
+```
+
+Parametrizes test with explicitly provided data.
+You can provide multiple examples for one test.
+In such a case, they are run sequentially.
+
+`example` can be used next to the [`given`](./given.md) cheatcode.
```

Comment:
Is protostar still going to fuzz when there is no `given` cheatcode?
If yes -> this explanation doesn't capture this behaviour
If not -> behaviour is inconsistent with standard fuzz tests

Reply:
> If not -> behaviour is inconsistent with standard fuzz tests

hmm you're quite right, but I wonder how to enforce doing parameterized testing without fuzzing in other way 🤔

Reply:
We want to support all 4 combinations:
1. standard tests
2. fuzz-only tests
3. parameterized-only tests
4. parameterized-and-then-fuzzed tests

Each of your ideas blocks some of these 4 combos. Current API allowed everything:

1. is trivial
2. use only `given`
3. use only `example`
4. use both `example` and `given` in single test

I think maybe this isn't actually so inconsistent. If you use a cheatcode, then you explicitly ask Protostar to do run specified mode, and you can combine modes naturally.

Reply:
> Is protostar still going to fuzz when there is no `given` cheatcode?

No, in such a case only `examples` will be taken under consideration.

Ok, so what's your suggestion here? We discussed somewhere in the comments that if we use `example`s without `given`, then we only want to have these examples executed, unlike when we do not use either of these cheatcodes - in this case, there's going to be a default strategy for `given` provided implicitly by protostar.

I'm going to add an explicit statement about that.

Reply:
[add an explicit statement about using example without given](https://github.com/software-mansion/protostar/pull/947/commits/b1acac1d73d7e7a36dbfa541ac3e53e124749a77)

Reply:
Sorry, I couldn't see the newest responses here.

It all makes sense to me anyway. Are there any more doubts here?

Reply:
I still have doubts being honest. We made a mistake designing initial verision of fuzzing, making it default strategy. No framework does that as far as I am aware. I would prefer to make fuzzing always explicit.

I created a issue to discuss that:

Approving, because the issue has nothing to do with your PR I believe.
Thanks for explicit statement, it makes things much cleaner.

Reply:
Resolving this conversation in favour of Maks's task